src/c4/ofpstream.cc

```diff
@@ -28,9 +28,11 @@ using namespace std;
  * \param filename Name of the file to which synchronized output is to be
  * written.
  */
-ofpstream::ofpstream(std::string const &filename) : std::ostream(&sb_) {
+ofpstream::ofpstream(std::string const &filename, ios_base::openmode const mode)
+    : std::ostream(&sb_) {
+  sb_.mode_ = mode;
```

Comment:
I think `mode_` should be added to the initializer list instead of being set in the ctor body.